src/js/server/lib/mailer.js

```diff
@@ -0,0 +1,156 @@
+/*
+
+  A common component for sending outgoing mail messages.  Used with the "forgot password" and "signup" APIs for now, but
+  will eventually be moved to a central package for use elsewhere.
+
+  Uses [nodemailer-smtp-transport](https://github.com/andris9/nodemailer-smtp-transport) at the moment.  All options
+  supported by that package can be configured using the `options.transportOptions` setting.
+
+  Mail content is created using `gpii-handlebars`.
+
+  // TODO:  Evaluate moving this to its own package, perhaps combining with gpii-mail-test.
+ */
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var nodemailer    = require("nodemailer");
+var smtpTransport = require("nodemailer-smtp-transport");
+
+var path        = require("path");
+var templateDir = path.resolve(__dirname, "../../src/templates");
+
+require("gpii-handlebars");
+
+fluid.registerNamespace("gpii.express.user.mailer");
```

Comment:
Even though it does not rely on express, it is part of this specific `gpii.express.user` package.  I could see evolving a `gpii-mailer` package, moving most of this code out, and keeping only the configuration here.